File: Exercicios_diarios/Revisão_intermediaria/Meio da lista vinculada.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        cont = 0
        aux = head
        while head:
            head = head.next
            cont += 1
        for i in range(int(cont / 2)):
            aux = aux.next
            i+=1
        return aux

File: Exercicios_diarios/Revisão_intermediaria/Meio da lista vinculada.cpp

struct ListNode
{
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};

class Solution
{
public:
    ListNode *middleNode(ListNode *head)
    {
        int cont = 0;
        ListNode *aux = head;
        while (head)
        {
            head = head->next;
            cont += 1;
        }
        for (int i = 0; i < cont / 2; i++)
            aux = aux->next;
        return aux;
    }
};